wsidicom/series/pyramids.py

```diff
@@ -109,6 +110,28 @@ def _group_instances_into_pyramids(
                 )
             else:
                 ext_depth_of_field = None
+
+            if instance.image_coordinate_system is not None:
+                existing_group_match = next(
+                    (
+                        (image_cs, ext_dof)
+                        for (image_cs, ext_dof), group in grouped_instances.items()
+                        if (
+                            all(
+                                instance.image_coordinate_system.origin_and_rotation_match(
+                                    inst.image_coordinate_system,
+                                    origin_threshold=settings.pyramids_origin_threshold,
+                                )
+                                for inst in group
+                            )
```

Comment:
If we sort the instances by size first, I think it is enough if we iterate and match over the keys?

Reply:
I like the idea to sort by size to resolve the indeterminacy. The idea currently is to ensure that the origin of the instance to insert into an existing group matches with all instances it contains already (max-diameter clustering). I think for that purpose the `group` (i.e. the dict value) is needed. An alternative implementation could be that the first instance is used define the cluster center. Then the maximum possible diameter effectively would become `2*settings.pyramids_origin_threshold`. Should I switch it to that?

Reply:
Thanks for the explanation. Lets keep your implementation with the change to run it on sorted instances.